crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -534,6 +504,96 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parses a complex literal pattern, given the `lhs` pattern and the `start`
+    /// position of the pattern.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at a `+` or `-` token.
+    ///
+    /// See: <https://docs.python.org/3/reference/compound_stmts.html#literal-patterns>
+    fn parse_complex_literal_pattern(
+        &mut self,
+        lhs: Pattern,
+        start: TextSize,
+    ) -> ast::PatternMatchValue {
+        let operator = if self.eat(TokenKind::Plus) {
+            Operator::Add
+        } else {
+            self.bump(TokenKind::Minus);
+            Operator::Sub
+        };
+
+        let lhs_value = if let Pattern::MatchValue(lhs) = lhs {
+            match &*lhs.value {
+                Expr::NumberLiteral(ast::ExprNumberLiteral {
+                    value: ast::Number::Complex { .. },
+                    ..
+                }) => {
+                    self.add_error(
+                        ParseErrorType::OtherError(
+                            "real number required in complex literal".to_string(),
+                        ),
+                        &lhs,
+                    );
+                }
```

Comment:
TODO: disallow (`-1j + 2j`) as well

Reply:
Do you plan to do this as part of this PR or as a follow up PR?